src/commands/server-admin/DeactivateCommand.tsx

```diff
@@ -0,0 +1,259 @@
+            {
+              sender: info.commandSender,
+              rule: null,
+            }
+          ))();
+    if (isError(deactivateResult)) {
+      return deactivateResult.elaborate(
+        `Failed to deactivate the user ${targetUser.toString()}`
+      );
+    } else {
+      return previewResult;
+    }
+  },
+});
+
+function renderUserDetails(preview: DeactivateUserPreview) {
+  return (
+    <details>
+      <summary>
+        Details for <code>{preview.targetUser}</code>
+      </summary>
+      <ul>
+        <li>
+          Creation date:{" "}
+          <code>
+            {new Date(preview.creation_timestamp).toLocaleDateString()}
+          </code>
+        </li>
+        <li>
+          Displayname:{" "}
+          {preview.displayname ? (
+            <code>{preview.displayname}</code>
+          ) : (
+            "None set"
+          )}
+        </li>
+        <li>
+          Purging messages:{" "}
+          {preview.isPurgingMessages ? <code>Yes</code> : <code>No</code>}
+        </li>
+      </ul>
+    </details>
+  );
+}
+
+DraupnirInterfaceAdaptor.describeRenderer(SynapseAdminDeactivateCommand, {
+  isAlwaysSupposedToUseDefaultRenderer: true,
+  confirmationPromptJSXRenderer(commandResult) {
+    if (isError(commandResult)) {
+      return Ok(undefined);
+    }
+    const preview = commandResult.ok;
+    return Ok(
+      <root>
+        You are about to deactivate the user <code>{preview.targetUser}</code>.
+        This will permanently deactivate the user and remove their access to the
+        homeserver.
+        {preview.isPurgingMessages ? (
+          <span>
+            Purging their messages will also cause their account to be used by
+            the homeserver to send hundreds of redaction events to remove
+            everything they have sent.
+          </span>
+        ) : (
+          <fragment></fragment>
+        )}
+        {renderUserDetails(preview)}
+      </root>
+    );
+  },
+  JSXRenderer(commandResult) {
+    if (isError(commandResult)) {
+      return Ok(undefined);
+    }
+    return Ok(
+      <root>
+        Successfully deactivated <code>{commandResult.ok.targetUser}</code>
+        {renderUserDetails(commandResult.ok)}
+      </root>
+    );
+  },
+});
+
+function renderPromptDeactivation(
+  targetUserID: StringUserID,
+  policyRule: LiteralPolicyRule
+): DocumentNode {
+  return (
+    <root>
+      The resident user <code>{targetUserID}</code> has been matched by a policy
+      rule that would cause their messages to be redacted if they joined a
+      protected room. Would you like to deactivate this user account and remove
+      all their messages?
+      <ul>{renderRuleSummary(policyRule)}</ul>
+    </root>
+  );
+}
+
+// FIXME:
+// hmm now would be a great time to include that feature where we can
+// run commands. but there probably isn't time for that right now.
+// we will just have to send a message, and then the confirmation prompt.
```

Comment:
Follow up issue here, we need to be able to create a prompt to run a command just like the confirmation prompt, but without adding additional arguments like the `--no-confirm` option.

I think this comment is more specifically asking for a way to embed the result and rendering of a command within another event that also includes the confirmation prompt which is probably just a bit silly.